### ISSUE TITLE
learning rates, nb of iter (global vs local), local line on the comparison graphs

### ISSUE THREAD

**Issue by eg10yx** [litian96/ditto#2]
Good afternoon,

I am a student at EPFL and I study your paper on Ditto for a project.

There are three aspect that are not clear to me in the Ditto algorithm.

1) are you using different learning rate for the global and the local optimization (ηg, ηl) and if yes how do choose them.
2) Where do you specify the number of global and local iteration (r,s) and how do you choose them.
3) On the performance graphs, to what correspond the 'local' constant line
<img width="728" alt="algo" src="https://user-images.githubusercontent.com/34946305/137898173-b9cab340-b8b5-48ec-a2e1-1f77516d7762.png">
<img width="577" alt="graph" src="https://user-images.githubusercontent.com/34946305/137898186-ebe5ba42-9eda-47d7-baae-c051651dbbdd.png">

Thank you very much for your repository,

Best regards,

Elie Graham

**Comment by litian96** [comment]
Hi,

For your first two questions, for each dataset, we tune the global learning rate on FedAvg based on validation data when there is no corruption (with completely clean training data), and use the same hyperparameters for all methods on that dataset. We use the same local and global learning rates in the experiments, if i remember correctly. 

The 'local' line corresponds to the method where we train separate local models for each device (no coordination). Since we don't need communication, it is a horizontal line.

**Comment by eg10yx** [comment]
Thank you for the clarifications
